src/utils.test.ts

```diff
@@ -208,66 +206,39 @@ describe('isHardened', () => {
   });
 });
 
-describe('stripHexPrefix', () => {
-  it('strips the hex prefix from a string', () => {
-    expect(stripHexPrefix('0x123')).toBe('123');
-    expect(stripHexPrefix('0x')).toBe('');
-  });
-
-  it('does not change a string without a prefix', () => {
-    expect(stripHexPrefix('123')).toBe('123');
-    expect(stripHexPrefix('')).toBe('');
-  });
-});
-
-describe('isValidHexString', () => {
-  it('returns true if the string is a valid hex string', () => {
-    expect(isValidHexString('0x123')).toBe(true);
-    expect(isValidHexString('0x0')).toBe(true);
-  });
-
-  it('returns false if the string is not a valid hex string', () => {
-    expect(isValidHexString('')).toBe(false);
-    expect(isValidHexString('0x')).toBe(false);
-    expect(isValidHexString('0x0g')).toBe(false);
-  });
-});
-
 describe('hexStringToBuffer', () => {
   it('returns the same buffer if a buffer is passed', () => {
-    const buffer = Buffer.from('123', 'hex');
+    const buffer = hexToBytes('123');
     expect(hexStringToBuffer(buffer)).toBe(buffer);
   });
 
   it('returns a buffer from a hex string', () => {
-    expect(hexStringToBuffer('1234')).toStrictEqual(Buffer.from('1234', 'hex'));
-    expect(hexStringToBuffer('0x1234')).toStrictEqual(
-      Buffer.from('1234', 'hex'),
-    );
+    expect(hexStringToBuffer('1234')).toStrictEqual(hexToBytes('1234'));
+    expect(hexStringToBuffer('0x1234')).toStrictEqual(hexToBytes('1234'));
   });
 
   it('throws if the string is not a valid hex string', () => {
-    expect(() => hexStringToBuffer('')).toThrow('Invalid hex string: "".');
-    expect(() => hexStringToBuffer('0x')).toThrow('Invalid hex string: "0x".');
```

Comment:
Do we support `0x` now?

Reply:
Yes. `hexToBytes('0x')` is equivalent to `new Uint8Array()`.